src/SFA.DAS.EmployerAccounts/Queries/GetSingleCohort/GetSingleCohortHandler.cs

```diff
@@ -0,0 +1,61 @@
+﻿using MediatR;
+using System.Threading.Tasks;
+using SFA.DAS.NLog.Logger;
+using SFA.DAS.Validation;
+using SFA.DAS.EmployerAccounts.Interfaces;
+using System.Linq;
+using SFA.DAS.HashingService;
+
+namespace SFA.DAS.EmployerAccounts.Queries.GetSingleCohort
+{
+    public class GetSingleCohortHandler : IAsyncRequestHandler<GetSingleCohortRequest, GetSingleCohortResponse>
+    {
+        private readonly IValidator<GetSingleCohortRequest> _validator;
+        private readonly ILog _logger;
+        private readonly ICommitmentV2Service _commitmentV2Service;        
+        private readonly IHashingService _hashingService;
+
+        public GetSingleCohortHandler(
+            IValidator<GetSingleCohortRequest> validator,
+            ILog logger,
+            ICommitmentV2Service commitmentV2Service,            
+            IHashingService hashingService)
+        {
+            _validator = validator;
+            _logger = logger;
+            _commitmentV2Service = commitmentV2Service;            
+            _hashingService = hashingService;
+        }
+
+
+        public async Task<GetSingleCohortResponse> Handle(GetSingleCohortRequest message)
+        {
+            var validationResult = _validator.Validate(message);
+
+            if (!validationResult.IsValid())
+            {
+                throw new InvalidRequestException(validationResult.ValidationDictionary);
+            }
+
+            long accountId = _hashingService.DecodeValue(message.HashedAccountId);
+
+            _logger.Info($"Getting Cohorts for account id {message.HashedAccountId}");
```

Comment:
Please remove unnecessary logging.  If there's an error there will be a stack trace.

Reply:
Its good to keep for now, so can  find out the exceptions in Kibana related to the Id

Reply:
Please remove.  It's unnecessary noise.  Exceptions will still be logged in Kibana .